specification/network/resource-manager/Microsoft.Network/stable/2021-05-01/firewallPolicy.json

```diff
@@ -567,6 +567,256 @@
           "nextLinkName": "nextLink"
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/firewallPolicies/{firewallPolicyName}/listIdpsSignatures": {
+      "post": {
+        "operationId": "FirewallPolicyIdpsSignatures_List",
+        "description": "Retrieves the current status of IDPS signatures for the relevant policy",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "firewallPolicyName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the Firewall Policy."
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/IDPSQueryObject"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success. The operation returns an IDPS Query Response",
+            "schema": {
+              "$ref": "#/definitions/QueryResults"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "query signature overrides": {
+            "$ref": "./examples/FirewallPolicyQuerySignatureOverrides.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/firewallPolicies/{firewallPolicyName}/signatureOverrides/default": {
```

Comment:
>"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/firewallPolicies/{firewallPolicyName}/signatureOverrides/default": [](http://example.com/codeflow?start=4&length=162)

Please add GET for listing signatureOverrides.

[https://armwiki.azurewebsites.net/api_contracts/guidelines/policy.html#plcy008-support-a-collection-get-and-individual-get-for-child-resource-types-even-singletons](https://armwiki.azurewebsites.net/api_contracts/guidelines/policy.html#plcy008-support-a-collection-get-and-individual-get-for-child-resource-types-even-singletons)

Reply:
There is a GET, it's in line: 726. There is also an example for that.

Reply:
You also need a GET for the type to list instances, even though this is a singleton. It will return a list of 1 item. Requires a GET on this path: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/firewallPolicies/{firewallPolicyName}/signatureOverrides

Reply:
done